WiserTaskScheduler/WiserTaskScheduler/Modules/CleanupWiserHistory/Services/CleanupWiserHistoryService.cs

```diff
@@ -76,31 +78,60 @@ public async Task<JObject> Execute(ActionModel action, JObject resultSets, strin
         await databaseConnection.ChangeConnectionStringsAsync(connectionStringToUse, connectionStringToUse);
         databaseConnection.ClearParameters();
 
-        var wiserItemsService = scope.ServiceProvider.GetRequiredService<IWiserItemsService>();
-        var tablePrefix = await wiserItemsService.GetTablePrefixForEntityAsync(cleanupWiserHistory.EntityName);
-
         var cleanupDate = DateTime.Now.Subtract(cleanupWiserHistory.TimeToStore);
-        databaseConnection.AddParameter("entityName", cleanupWiserHistory.EntityName);
         databaseConnection.AddParameter("cleanupDate", cleanupDate);
-        databaseConnection.AddParameter("tableName", $"{tablePrefix}{WiserTableNames.WiserItem}");
 
-        var historyRowsDeleted = await databaseConnection.ExecuteAsync($"""
+        var historyEntriesToDelete = new List<ulong>();
+
+        if (!String.IsNullOrWhiteSpace(cleanupWiserHistory.EntityName))
+        {
+            var wiserItemsService = scope.ServiceProvider.GetRequiredService<IWiserItemsService>();
+            var tablePrefix = await wiserItemsService.GetTablePrefixForEntityAsync(cleanupWiserHistory.EntityName);
+
+            databaseConnection.AddParameter("entityName", cleanupWiserHistory.EntityName);
+            databaseConnection.AddParameter("tableName", $"{tablePrefix}{WiserTableNames.WiserItem}");
 
-                                                                        DELETE history.*
-                                                                        FROM {tablePrefix}{WiserTableNames.WiserItem} AS item
-                                                                        JOIN {WiserTableNames.WiserHistory} AS history ON history.item_id = item.id AND history.tablename LIKE CONCAT(?tableName, '%') AND history.changed_on < ?cleanupDate
-                                                                        WHERE item.entity_type = ?entityName
-                                                                        """);
+            var dataTable = await databaseConnection.GetAsync($"""
+                                                               SELECT CAST(history.id AS UNSIGNED) AS id
+                                                               FROM {tablePrefix}{WiserTableNames.WiserItem} AS item
+                                                               JOIN {WiserTableNames.WiserHistory} AS history ON history.item_id = item.id AND history.tablename LIKE CONCAT(?tableName, '%') AND history.changed_on < ?cleanupDate
+                                                               WHERE item.entity_type = ?entityName
+                                                               """);
 
-        historyRowsDeleted += await databaseConnection.ExecuteAsync($"""
+            historyEntriesToDelete.AddRange(from DataRow row in dataTable.Rows select row.Field<ulong>("id"));
```

Comment:
Our .NET code conventions call for the usage of the member extension methods instead of the SQL-style query expressions:
> When using LinQ, prefer member extension methods over SQL-style LINQ keywords - e.g. prefer `myList.Where(x)` to `myList where x`.

In this case, the `Select` method could be used.